scout/server/blueprints/institutes/controllers.py

```diff
@@ -619,15 +622,37 @@ def update_variant_genes(store, variant_obj, genome_build):
         gene_obj["hgnc_symbol"] = gene_caption["hgnc_symbol"]
         gene_obj["description"] = gene_caption["description"]
 
-        gene_symbols.add(gene_obj["hgnc_symbol"])
+        gene_symbols.append(gene_obj["hgnc_symbol"])
+
+        # ensure annotations are populated
+        functional_annotations.append(gene_obj.get("functional_annotation"))
+        region_annotations.append(gene_obj.get("region_annotation"))
 
         # gather HGVS info from gene transcripts
-        (hgvs_nucleotide, hgvs_protein) = get_hgvs(gene_obj)
+        (canonical_transcript, hgvs_nucleotide, hgvs_protein) = get_hgvs(gene_obj)
+
+        canonical_transcripts.append(canonical_transcript)
         hgvs_c.append(hgvs_nucleotide)
         hgvs_p.append(hgvs_protein)
 
-    if len(gene_symbols) == 1:
-        variant_obj["hgvs"] = hgvs_str(list(gene_symbols), hgvs_p, hgvs_c)
+    variant_obj["hgvs"] = hgvs_str(gene_symbols, canonical_transcripts, hgvs_p, hgvs_c)
+    if len(functional_annotations) == 1:
```

Comment:
It would look cleaner if you only had the loop here, doesn't matter if it repeats the gene

Reply:
Hm, but that is redundant. Let me flesh out that other function I thought about first instead. That will save some repetitions while still making the output less messy.